File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-opposite-direction/4sum/code/python/4sum_sol1.py
class Solution:
    def fourSum(self, nums, target):
        nums.sort(); result = set()
        for i in range(len(nums)-3):
            for j in range(i+1,len(nums)-2):
                for k in range(j+1,len(nums)-1):
                    for l in range(k+1,len(nums)):
                        if nums[i]+nums[j]+nums[k]+nums[l]==target:
                            result.add((nums[i],nums[j],nums[k],nums[l]))
        return [list(t) for t in result]

File: src/data/courses/dsa/two-pointers-sliding-window/two-pointers-opposite-direction/4sum/code/cpp/4sum_sol1.cpp
class Solution {
public:
    // Brute Force: O(n⁴)
    vector<vector<int>> fourSum(vector<int>& nums, int target) {
        sort(nums.begin(), nums.end());
        set<vector<int>> result;
        int n = nums.size();
        for (int i=0;i<n-3;i++) for (int j=i+1;j<n-2;j++)
            for (int k=j+1;k<n-1;k++) for (int l=k+1;l<n;l++)
                if ((long)nums[i]+nums[j]+nums[k]+nums[l]==target)
                    result.insert({nums[i],nums[j],nums[k],nums[l]});
        return vector<vector<int>>(result.begin(), result.end());
    }
};